include/url.h

```diff
@@ -79,6 +79,22 @@ namespace Url
             , fragment_(other.fragment_)
             , userinfo_(other.userinfo_) { }
 
+        /**
+         * Take on the value of the other URL.
+         */
+        Url& assign(const Url& other);
```

Comment:
By which I mean, the Cython implementation uses the PIMPL idiom, but we cannot do this:

```
# Can't dereference in the normal way
*self.ptr = *other.ptr

# Instead we have to use 'dereference', but now this is nonsensical:
dereference(self.ptr) = dereference(other.ptr)

# Also, we can't access the `operator=` member on self.ptr, AFAIK
```